### PR TITLE
iOS Basic Text Input

### PR DESCRIPTION
**Pull request by danwalmsley**
## What does the pull request do?

Updated the IME apis to support InputOptions.
Adds attached properties to control input options.
Adds fully working standard input method support for ios.

## What is the current behavior?

ios input is based on hacks.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Obsoletions / Deprecations

- TextInputOptionsQuery event was removed.
- TextInputContentType.Phone was removed.

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0019165-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Review by kekekeks — CHANGES_REQUESTED** [review]

File: src/Avalonia.Input/TextInput/InputMethodManager.cs [hunk 1]
```diff
@@ -91,12 +92,15 @@ public void SetFocusedElement(IInputElement? element)
...
             var inputMethod = (element?.VisualRoot as ITextInputMethodRoot)?.InputMethod;
-            if (_im != inputMethod)
-                _im?.SetActive(false);
-
+            
             _im = inputMethod;
 
             TryFindAndApplyClient();
+
+            if (_im != inputMethod)
```
Comment (kekekeks) [inline src/Avalonia.Input/TextInput/InputMethodManager.cs:100]: This logic already exists in Client property setter, I think

File: src/Avalonia.Input/TextInput/InputMethodManager.cs [hunk 2]
```diff
@@ -91,12 +92,15 @@ public void SetFocusedElement(IInputElement? element)
             _focusedElement = element;
 
             var inputMethod = (element?.VisualRoot as ITextInputMethodRoot)?.InputMethod;
-            if (_im != inputMethod)
-                _im?.SetActive(false);
-
+
```
Comment (kekekeks) [inline src/Avalonia.Input/TextInput/InputMethodManager.cs:95]: It was needed here to forcefully deactivate IM

Comment (danwalmsley) [reply src/Avalonia.Input/TextInput/InputMethodManager.cs]: fixed.

File: src/Avalonia.Input/TextInput/InputMethodManager.cs (hunk 1 above)
Comment (danwalmsley) [reply src/Avalonia.Input/TextInput/InputMethodManager.cs:96]: removed.

File: src/iOS/Avalonia.iOS/AvaloniaUIResponder.cs [hunk 3]
```diff
@@ -0,0 +1,119 @@
+using Foundation;
```
[the remark this answers is not in the document]
Comment (danwalmsley) [reply src/iOS/Avalonia.iOS/AvaloniaUIResponder.cs]: renamed.

File: src/Avalonia.Input/TextInput/InputMethodManager.cs [hunk 4]
```diff
@@ -35,21 +35,22 @@ public TextInputMethodManager()
...
-                    {
-                        RoutedEvent = InputElement.TextInputOptionsQueryEvent
-                    };
-                    _focusedElement?.RaiseEvent(optionsQuery);
+
                     _im?.Reset();
-                    _im?.SetOptions(optionsQuery);
-                    _transformTracker?.SetVisual(_client?.TextViewVisual);
+                    
+                    if (_focusedElement is StyledElement target)
```
Comment (kekekeks) [inline src/Avalonia.Input/TextInput/InputMethodManager.cs:41]: We need to reset options if none are provided for some reason

**Review by kekekeks — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0019210-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Pull request merged**